tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -141,699 +186,857 @@ public void testPrintVersion(String quorum) {
-
-        Properties customProps = new Properties();
-        // create a consumer group that never commits offsets
-        customProps.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
-        // run one consumer in the group consuming from a single-partition topic
-        addConsumerGroupExecutor(1, TOPIC, GROUP, RangeAssignor.class.getName(), Optional.empty(), Optional.of(customProps), false, groupProtocol);
-
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", GROUP};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+    private void createTopic(String topic) {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            Assertions.assertDoesNotThrow(() -> admin.createTopics(Collections.singletonList(new NewTopic(topic, 1, (short) 1))).topicId(topic).get());
+        }
+    }
 
-        TestUtils.waitForCondition(() -> {
-            Entry<Optional<ConsumerGroupState>, Optional<Collection<PartitionAssignmentState>>> groupOffsets = service.collectGroupOffsets(GROUP);
+    private void createTopic(String topic, int numPartitions) {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            Assertions.assertDoesNotThrow(() -> admin.createTopics(Collections.singletonList(new NewTopic(topic, numPartitions, (short) 1))).topicId(topic).get());
+        }
+    }
 
-            Predicate<PartitionAssignmentState> isGrp = s -> Objects.equals(s.group, GROUP);
+    private void deleteConsumerGroups(Collection<String> groupIds) {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            Assertions.assertDoesNotThrow(() -> admin.deleteConsumerGroups(groupIds).all().get());
+        }
+    }
 
-            boolean res = groupOffsets.getKey().map(s -> s.equals(ConsumerGroupState.STABLE)).orElse(false) &&
-                groupOffsets.getValue().isPresent() &&
-                groupOffsets.getValue().get().stream().filter(isGrp).count() == 1;
+    private void deleteTopic(String topic) {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            Assertions.assertDoesNotThrow(() -> admin.deleteTopics(Collections.singletonList(topic)).topicNameValues().get(topic).get());
+        }
+    }
 
-            if (!res)
-                return false;
+    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, Set<TopicPartition> topicPartitions, Map<String, Object> customConfigs) {
+        Map<String, Object> configs = composeConfigs(
```

Comment:
`return consumerGroupClosable(protocol, groupId, topicName, customConfigs, 1);`?

Reply:
No, we can't use this, because one is `topicPartitions`, and another is `topicName`.

Reply:
There are two similar methods.
```java
    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, String topicName, Map<String, Object> customConfigs) {
        Map<String, Object> configs = composeConfigs(
                groupId,
                protocol.name,
                customConfigs
        );
        return ConsumerGroupCommandTestUtils.buildConsumers(
                1,
                false,
                topicName,
                () -> new KafkaConsumer<String, String>(configs)
        );
    }

    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, String topicName, Map<String, Object> customConfigs, int numConsumers) {
        Map<String, Object> configs = composeConfigs(
                groupId,
                protocol.name,
                customConfigs
        );
        return ConsumerGroupCommandTestUtils.buildConsumers(
                numConsumers,
                false,
                topicName,
                () -> new KafkaConsumer<String, String>(configs)
        );
    }
```

What I mentioned is the first method

Reply:
Updated it. Thanks.